src/components/editor/document-renderer-pane/scrolling-document-render-pane.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react'
+import { LineMarkerPosition } from '../../markdown-renderer/types'
+import { ScrollProps, ScrollState } from '../scroll/scroll-props'
+import { findLineMarks } from '../scroll/utils'
+import { DocumentRenderPane, DocumentRenderPaneProps } from './document-render-pane'
+
+export const ScrollingDocumentRenderPane: React.FC<DocumentRenderPaneProps & ScrollProps> = ({
```

Comment:
This component does so much more then just scrolling.. But right know I don't have an idea for a better structure.

Reply:
what does it do besides scrolling? It seems to me that's more or less it purpose…